main/lsp/requests/code_action.cc

```diff
@@ -0,0 +1,64 @@
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/lsp.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+LSPResult LSPLoop::handleTextDocumentCodeAction(unique_ptr<core::GlobalState> gs, const MessageId &id,
+                                                const CodeActionParams &params) {
+    auto response = make_unique<ResponseMessage>("2.0", id, LSPMethod::TextDocumentCodeAction);
+
+    if (!opts.lspQuickFixEnabled) {
+        response->error = make_unique<ResponseError>(
+            (int)LSPErrorCodes::InvalidRequest, "The `Quick Fix` LSP feature is experimental and disabled by default.");
+        return LSPResult::make(move(gs), move(response));
+    }
+
+    vector<unique_ptr<CodeAction>> result;
+
+    prodCategoryCounterInc("lsp.messages.processed", "textDocument.codeAction");
+
+    core::FileRef file = uri2FileRef(params.textDocument->uri);
+    vector<shared_ptr<core::File>> files;
+    // TODO(sushain): is there another way to do this? maybe a helper I'm missing?
+    files.push_back(make_shared<core::File>(string(file.data(*gs).path()), string(file.data(*gs).source()),
+                                            core::File::Type::Normal));
+    // TODO(sushain): why does tryFastPath(move(gs), {}, files) not work (when files is refs)?
+    auto run = tryFastPath(move(gs), files);
+    for (auto &e : run.errors) {
+        if (!e->isSilenced && e->loc.file() == file && !e->autocorrects.empty() &&
+            cmpRanges(*loc2Range(*run.gs, e->loc), *params.range) == 0) {
+            vector<unique_ptr<TextEdit>> edits;
+            edits.reserve(e->autocorrects.size());
+            for (auto &a : e->autocorrects) {
+                edits.emplace_back(make_unique<TextEdit>(loc2Range(*run.gs, a.loc), a.replacement));
+            }
+
+            vector<unique_ptr<TextDocumentEdit>> documentEdits;
+            documentEdits.emplace_back(make_unique<TextDocumentEdit>(
+                make_unique<VersionedTextDocumentIdentifier>(params.textDocument->uri, JSONNullObject()), move(edits)));
```

Comment:
One of these days we'll actually track document versions. Can you put a `TODO: Document version` comment in here so we don't forget?